app/models/step_by_step_page.rb

```diff
@@ -72,6 +72,10 @@ def links_checked?
     steps.map(&:link_report?).any?
   end
 
+  def discard_notes
+    InternalChangeNote.where(step_by_step_page_id: self.id, edition_number: nil).delete_all
```

Comment:
Does using `internal_change_notes.where(edition_number: nil).delete_all` work here? We shouldn't need to pass in the step_by_step_id as it should already know who it's children are.

Reply:
Yes, that works perfectly. Thank you